components/zeekopenbsm/src/openbsmconsumer.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <bsm/libbsm.h>
+#include <set>
+#include <vector>
+#include <zeek/iopenbsmconsumer.h>
+
+namespace zeek {
+class OpenbsmConsumer final : public IOpenbsmConsumer {
+public:
+  /// \brief Destructor
+  ~OpenbsmConsumer() override;
+
+  /// \brief Returns a list of processed events
+  /// \param event_list Where the event list is stored
+  /// \return A Status object
+  Status getEvents(EventList &event_list) override;
+
+private:
+  struct PrivateData;
+  std::unique_ptr<PrivateData> d;
+  // audit pipe handler
+  FILE *audit_pipe{nullptr};
+
+  /// list of subscribed events
+  std::set<size_t> event_filter_list;
+
+  /// \brief Constructor
+  OpenbsmConsumer(IZeekLogger &logger, IZeekConfiguration &configuration);
+
+  /// \brief extract header from openbsm token and populate event
+  static void extractHeader(Event &event, tokenstr_t tok);
+  /// \brief extract subject from openbsm token and populate event
+  static void extractSubject(Event &event, tokenstr_t tok);
+  /// \brief extract return from openbsm token and populate event
+  static void extractReturn(Event &event, tokenstr_t tok);
+  /// \brief extract socket-inet from openbsm token and populate event
+  static void extractSocketInet(Event &event, tokenstr_t tok);
```

Comment:
No specific reason. Can this create some issue? What do you suggest?

Reply:
Out parameters are harder to understand than simply returning a value since they e.g., force callers to default-construct a value which is only set up later. This can make it e.g., less obvious whether a value is set on all paths.

I'd suggest changing all the functions here to take a `tokenstr_t` and to return an `Event` by value. `populateSocketEvent` take a `const vector<tokenstr_t>&` should return an `Event` as well (the status is not needed).

Reply:
In the latest commit, these functions return `Status` object to show if the extraction was successful or not so that we can log in a failure scenario. Let me know what you think.